### ISSUE TITLE
Investigating generic parameters assigned to DrugBank Molecules

### ISSUE THREAD

**Issue by bannanc** [open-forcefield-group/smirnoff99Frosst#14]
In connection with the Mobley Group undergraduate project [off-ffcompare](https://github.com/MobleyLab/off-ffcompare) I've been looking at how SMIRNOFF99Frosst types the molecules in DrugBank. Below I'll list in detail, what I'm finding, but the most surprising seems to be that for torsions some ethers seem to get assigned the generic ('t1') parameter. I'm haven't spent much time investigating why this might be happening (i.e. I haven't opened the ffxml or the "smirffishFrcmod" files yet)

***Non-Bonding***
We're missing Boron and Silicon (and helium...) those are the three atoms that get 'n1' `[*:1]` the generic non-bonding parameter. I don't know why, but Helium as a single atom is in the DrugBank database. There are a total of 19 molecules with at least one of those atoms

***Bonds***
Excluding Silicon and boron, here are the types of functional groups being assigned 'b1' `[*:1]~[*:2]`
There are 81 molecules with at least one generic bond. Most have a carbonyl like group with sulfur (C=S)

Here's the full list of bonds we seem to be missing
* Sulfur
    - S=0
         - in a group with the shape *CX4-S(=O)=O the sulfur has an unpaired electron
    - C=S
        - Most of these were like a carbonyl group, but one was C=S-N
    - P-S

* Phosphorous
    - P-Cl
    - P-H

* Nitrile Carbon
    - C(nitrile)#N+1-CX4
    - C(nitrile)-S
    - C(nitrile)-I

* Weird Cases
    - N-Br
         - I don't think this is common, but we should probably catch it
    - in CO2 the C=0 bonds aren’t typed?
         - I'm guessing this is because carbonyl carbons are probably written as [#6X3]=[#8] and the carbon dioxide carbon isn't trivalent, but I'm not entirely sure
    - in O2 the O=O bond, but that is literately the only molecule where you’ll see that bond

***Angles***
There aren't many here, only 3 molecules with the generic angle 'a1' `[*:1]~[*:2]~[*:3]`, I once again skipped any molecules with Boron or Silicon. 

* Sulfur
    - [#6X3]=[#16X2]-[#7]
        - Same as the weird C=S bond case above
    - [#8]=[#16X2]-[#6X4]
        - This is an oxygen double bonded to a divalent sulfur bound to a tetrahedral carbon
* Nitrogen 
    - [#6X2]#[#7+1]-[#6X4]
         - So this is a nitrile group with a positive charge on the nitrogen so it has the shape R-C#N+-R 

***Torsions***
There are 2499 molecules of the 6122 being tested from the DrugBank molecules database that are assigned at least one generic torsion 't1' `[*:1]~[*:2]~[*:3]~[*:4]`. That is a ridiculous number!!!

A LOT of them are ethers where the torsion getting the generic is in the form 
[#1:1]-[#6X4]-[#8X2]-[#6]
hydrogen - tetravalent carbon - oxygen - carbon
the last carbon is sometimes tetrahedral and sometimes aromatic. We had these torsions even in the AlkEthOH SMIRFF so I don't know why they aren't being typed now. I'm going to look more closely at some of the individual molecules later this week.

**Comment by davidlmobley** [comment]
@cbayly13 - please see above in case it affects any of the cases you are looking at. Are you checking whether any of the cases you are looking at get generics? You can use the forcefield labeler for this. 

@bannanc - we should look at this ASAP. Not sure I'll have time this week but I'd like to. These should be easy fixes by checking the relevant smirffish-frcmod files Christopher created.

Regarding the ether case you mention at the bottom, recall that SMIRFF99Frosst is entirely separate from AlkEthOH (though I've got a personal derivative version of it that will give equivalent energies on AlkEthOH) so just because something is "right" in AlkEthOH doesn't mean it will be here. However, the case you mention does sound concerning. 

Could you attach molecules getting generic parameters to this issue?

**Comment by bannanc** [comment]
There are 2499 molecules, here are two of them, I saved in my notes yesterday:

![332529a1-b245-48e6-81c9-60c1efde8867](https://cloud.githubusercontent.com/assets/11968541/23478858/7e3caca8-fe77-11e6-85a6-6499bf0beb5c.png)

![a3a7ce81-50b1-4d26-be55-91f26fc1d1c4](https://cloud.githubusercontent.com/assets/11968541/23478837/703f7d2e-fe77-11e6-9baf-d23f39e8daba.png)

I'm going to try the same molecules with Daisy's mol2 files that have 3d coordinates and see if we get the same problem. I am having a hard time believing the ethers because I get the same parameters with smirky on AlkEthOH using the AlkEthOH_smirff and this smirff that's why I found it surprising. 

I'll make checking this out a priority today.

**Comment by bannanc** [comment]
>  I am having a hard time believing the ethers because I get the same parameters with smirky on AlkEthOH using the AlkEthOH_smirff and this smirff that's why I found it surprising.

@davidlmobley and @cbayly13 

I need to bite my tongue here, I thought I ran AlkEthOH with smirff99Frosst, but I guess I only used the Frosst_AlkEtOH.ffxml file. So this missing ether torsion might explain why I was consistently getting 100% on AlkEthOH torsions with smirky, but couldn't on PhEthOH (where I was using smirff99Frosst). 

I'm looking at the smirffishFrcmod.parm99Frosst.txt file in the Dihedrals centered on a C-O bond there are none that would fit torsions that have H-C-O-C

I just ran our find generic parameters notebook on the AlkEthOH molecules. There are 30 (of 42) that get the generic torsion. All the highlighted torsions are in the shape H-C-O-C 
Here's the complete list of molecules with at least one 't1' assigned:

```
AlkEthOH_r0         C1C(C(O1)O)O
AlkEthOH_r19        C1CC(CCCOC1)O
AlkEthOH_r43        C1CCC(CCOCC1)O
AlkEthOH_r17        C1CC(CC(OC1)O)O
AlkEthOH_r49        C1CCC(OCC1)O
AlkEthOH_r29        C1CC(OCCCOC1)O
AlkEthOH_r47        C1CCC(OC(CC1)O)O
AlkEthOH_r11        C1CC(C(COC1)O)O
AlkEthOH_r21        C1CC(COC1)O
AlkEthOH_r37        C1CCC(C(OCC1)O)O
AlkEthOH_r28        C1CC(OC1O)O
AlkEthOH_r48        C1CCC(OC(OCC1)O)O
AlkEthOH_r26        C1CC(OC(OC1)O)O
AlkEthOH_r12        C1CC(C(OC1)O)O
AlkEthOH_r44        C1CCC(COCC1)O
AlkEthOH_r27        C1CC(OC1)O
AlkEthOH_r41        C1CCC(CC(OCC1)O)O
AlkEthOH_r22        C1CC(COCOC1)O
AlkEthOH_r30        C1CC(OCCOC1)O
AlkEthOH_r46        C1CCC(OC(C1)O)O
AlkEthOH_r24        C1CC(OC(C1)O)O
AlkEthOH_r7         C1C(OC1O)O
AlkEthOH_r4         C1C(COC1O)O
AlkEthOH_r31        C1CC(OCOC1)O
AlkEthOH_r3         C1C(CO1)O
AlkEthOH_r6         C1C(OC(O1)O)O
AlkEthOH_r8         C1C(OCO1)O
AlkEthOH_r16        C1CC(CC(COC1)O)O
AlkEthOH_r20        C1CC(CCOC1)O
AlkEthOH_r25        C1CC(OC(COC1)O)O
```

**Comment by davidlmobley** [comment]
@bannanc - I suspect you'll be able to find a simple fix here if you look at the smirffish-frcmod files.

However, I'm puzzled as to why I didn't already catch this, as I went through AlkEthOH (while you were out for orals) and made sure smirff99Frosst could either (a) reproduce AMBER energetics for all of the compounds or (b) the differences were cases I understood which introduced additional generality and were added deliberately by Chris. There were no generic torsions applied to AlkEthOH when I was done, IIRC (nor even when I started, I believe).

Can you confirm where exactly you got the file you are presently using? Is it the current one from this repo?

I should probably dig out my notebooks/notes from where I was doing this testing. 

I suppose there's also the remote possibility that I somehow botched something in the final stages of testing and did something crazy like accidentally deleting a torsion which resulted in generics being applied -- but since I was looking at reproduction of energetics my testing should have caught that.

I can't dig in right now, so please just verify you're using the correct file. It might be also helpful to look at the commit history here.

**Comment by bannanc** [comment]
> Can you confirm where exactly you got the file you are presently using? Is it the current one from this repo?

@davidlmobley   I'm using /smarty/utilities/convert_frosst/parameter_usage.ipynb 
the only thing I changed was which molecule set I loaded in. I diff'd the smirff99Frosst.ffxml in that directory with the one in this repo and they are identical. 

I'll take a look at the commit history on the smirffishFrcmod file and see if there was ever a torsion for this that was removed. Assuming it was never there I'll find the Frcmod file and see if I can easily make a new torsion parameter.

**Comment by davidlmobley** [comment]
Thanks. Keep me posted. I'll obviously have to dig back and see why I did not catch this in the energy comparisons...

**Comment by bannanc** [comment]
@davidlmobley 

I found something that might have caused the differences in what I found here and your tests. 

If I use `smirffishFrcmod.parm99Frosst_AlkEthOH.txt` to make a ffxml file there are no generic torsions with AlkEthOH. There is a more generic ether torsion here that is missing from `smirffishFrcmod.parm99Frosst.txt`

The torsion `[a,A:1]-[#6X4:2]-[#8X2:3]-[!#1:4]` is in the AlkEthOH Frcmod file, but not the one actually used to make smirff99frosst. 

`[a,A:1]-[#6X4:2]-[#8X2:3]-[!#1:4]` is  [any atom]-[tetrahedral carbon]-[divalent oxygen]-[not hydrogen]

**Comment by bannanc** [comment]
# History of smirffishFrcmod.parm99Frosst.txt

I began by working backward to see if I could find if there used to be a more general ether torsion in smirffishFrcmod.parm99Fross.txt that was removed. 

* [commit cb5d5c71b2f677d3413acdd190f54775b36dbaef](https://github.com/open-forcefield-group/smarty/commit/cb5d5c71b2f677d3413acdd190f54775b36dbaef#diff-36e1d8cfcdea7e0fdb8d3bc89f6982bb)
Has torsions not in the current version, I didn't see where they were removed so I'm looking at the newer commits again.
```
[*:1]-[#6X4:2]-[#8X2H0:3]-[*:4]      1  0.383    0.0  -3.0  parm99 CT-CT-OS-CT
[*:1]-[#6X4:2]-[#8X2H0:3]-[*:4]      1  0.100  180.0   2.0  parm99 CT-CT-OS-CT 
```

* [commit 51c6b6ca968d548a4617d9f2517d1c57fc9fca44](https://github.com/open-forcefield-group/smarty/commit/51c6b6ca968d548a4617d9f2517d1c57fc9fca44#diff-36e1d8cfcdea7e0fdb8d3bc89f6982bb)
This is the next commit on the smirffishFrcmod file, the more general ether torsion was replaced with one specifically for C-C-O-C instead of *-C-O-* so the lines above became:
```
[#6X4:1]-[#6X4:2]-[#8X2H0:3]-[#6X4:4]      1  0.383    0.0  -3.0  parm99 CT-CT-OS-CT
[#6X4:1]-[#6X4:2]-[#8X2H0:3]-[#6X4:4]      1  0.100  180.0   2.0  parm99 CT-CT-OS-CT
```

However, I would assume that H-C-O-C and C-C-O-C should not have the same torsion parameters so there was probably an energetic issue before you made the generic torsion too specific.

**Comment by bannanc** [comment]
# Smirffish_archive 

Initially I started looking at `frcmod.parm_Frosst`, but then I remembered all of Christopher's intermediate files that we saved (for this purpose) so I thought I would start there. 

I'm currently looking at `smirffishFrcmod.Torsions.parm99withParmFrosst.txt`. Here is the ether section from that file, it looks the same as the pre commit 51c6b6ca968d548a4617d9f2517d1c57fc9fca44 smirffish file. 
```
# O-Csp3
[*:1]-[#6X4:2]-[#8X2H1:3]-[#1:4]     1  0.160    0.0  -3.0  parm99 X_-CT-OH-X  JCC,7,(1986),230
[*:1]-[#6X4:2]-[#8X2H1:3]-[#1:4]     1  0.250    0.0   1.0  parm99 HO-OH-CT-CT  Junmei et al, 1999
[*:1]-[#6X4:2]-[#8X2H0:3]-[*:4]      1  0.383    0.0  -3.0  parm99 CT-CT-OS-CT
[*:1]-[#6X4:2]-[#8X2H0:3]-[*:4]      1  0.100  180.0   2.0  parm99 CT-CT-OS-CT
[#6X4:1]-[#6X4:2]-[#8X2:3]-[#6X3:4]  1  0.383    0.0  -3.0  parm99 CT-CT-OS-C  Junmei et al, 1999
[#6X4:1]-[#6X4:2]-[#8X2:3]-[#6X3:4]  1  0.800  180.0   1.0  parm99 CT-CT-OS-C  Junmei et al, 1999
[#6X4:1]-[#8X2:2]-[#6X4:3]-[#8X2:4]  1  0.100    0.0  -3.0  parm99 CT-OS-CT-OS  Junmei et al, 1999
[#6X4:1]-[#8X2:2]-[#6X4:3]-[#8X2:4]  1  0.850  180.0  -2.0  parm99 CT-OS-CT-OS  Junmei et al, 1999
[#6X4:1]-[#8X2:2]-[#6X4:3]-[#8X2:4]  1  1.350  180.0   1.0  parm99 CT-OS-CT-OS  Junmei et al, 1999
[#6X4:1]-[#8X2:2]-[#6X4:3]-[#7X3:4]  1  0.383    0.0  -3.0  parm99 CT-OS-CT-N*  parm98.dat, TC,PC,PAK
[#6X4:1]-[#8X2:2]-[#6X4:3]-[#7X3:4]  1  0.650    0.0   2.0  parm99 CT-OS-CT-N*  Piotr et al.
```
This is followed by the O-Csp3cyclopropyl and O-Csp2 sections, that I'm not worried about right now. 

# frcmod.parm_Frosst

This would have assigned H-C-O-C and C-C-O-C torsions the same values though, which I'm still skeptical of so I also looked at the original frcmod file. Here is a list of the relevant torsions I've found so far, with line numbers for context, they are not all near each other in this file. 
extra space when lines weren't next to each other.
```
X -OS-C -X      2 5.000  180.000 2 half ff94 X-C-N-X

X -OS-CA-X      1 1.350  180.000 2 RHF/6-31G(d,p) Dec 12 1997

X -OS-CB-X      2 6.000  180.000 2 ff94 X-CW-NA-X

X -OS-CR-X      2 4.800  180.000 2 ff94 X-CV-NB-X

X -OS-CW-X      2 6.000  180.000 2 ff94 X-CW-NA-X

X -OS-NB-X      1 0.000    0.000 1

CT-OS-CA-CA     1 1.350 180.000 2 RHF/6-31G(d,p) Dec 12 1997

CA-OS-CJ-CJ     1 0.000   0.000 1 guess; 26 April 2006
CA-OS-CJ-H1     1 0.000   0.000 1 guess; 26 April 2006

CA-OS-CM-C      1 0.000    0.000 1 guess
CT-OS-CM-C      1 0.000    0.000 1 guess
CA-OS-CM-CM     1 0.000    0.000 1 guess
CT-OS-CM-CM     1 0.000    0.000 1 guess
CW-OS-CM-CM     1 1.350  180.000 2 From X-CA-OS-X
SO-OS-CM-CM     1 0.000    0.000 1 guess
CA-OS-CM-CT     1 0.000    0.000 1 guess
CT-OS-CM-CT     1 0.000    0.000 1 guess
CA-OS-CM-H4     1 1.350  180.000 2 From X-CA-OS-X
CT-OS-CM-H4     1 0.000    0.000 1 guess
SO-OS-CM-H4     1 0.000    0.000 1 guess

CA-OS-CP-CA     1 0.000    0.000 1 guess for rigid systems; 13 April 2005
CA-OS-CP-CP     1 0.000    0.000 1 guess for rigid systems; 13 April 2005
```

**Comment by davidlmobley** [comment]
@bannanc -- the `smirffishFrcmod.parm99Frosst_AlkEthOH.txt` file was created by me in the process of making a smirff99Frosst derivative which would reproduce AMBER energetics exactly on AlkEthOH. It is not from Bayly. Let me drill down and see if I can see what I said about this particular SMIRKS.

**Comment by davidlmobley** [comment]
@bannanc - I have these notes concerning bugs found in smirff99Frosst; however, I thought I'd fixed them all:

> ### Bugs with smirff99frosst
> - c10 fails on account of 2-6 bond length, a C-O bond in C(C(O)OCO)O, a diol. That's the [#6X4]-[#8H0] bond; it seems to be picking up a [#6X4:1]-[#8X2] parameter; there is a [#6X3:1]-[#8X2H0] that does not get applied here (because it's C3) but no 6X4 suitable pattern. It looks like a failure to port the CT-OS parameter. Bizarrely, though, I can't find this in any of the files... Ah, perhaps he deleted it by mistake, as it's in his AlkEthOH frcmod file. This is a parm@frosst difference from parm99. FIXED. It's possible he removed this deliberately, though there is no record of that and it comes from a crystal, the files say.
> - After that, get torsion 7,0,3,12 with differing numbers of terms (H1-C1-O1-H6), HC-CT-OH-HO. OK, there are going to be a lot of funky torsions because of all of the AMBER bugs; re-add these. That did not do it; turns out this specific one was caused by a bug in his SMARTS for HO-OH-CT-CT, now fixed.
> - C-O generic torsion was not broad enough (didn't get X-OS-CT-X as in AlkEthOH).
> - Fix overly broad CT-CT-OS-CT smirks (had generics in first and last positions)
> - Fix omitted HO-OH-CT-CT torsional term

Still drilling down.

**Comment by davidlmobley** [comment]
So, @bannanc - to be slightly more specific, this line from the smirffishFrcmod_AlkEthOH:

> `[a,A:1]-[#6X4:2]-[#8X2:3]-[!#1:4]    3  1.150    0.0  -3.0  Frosst X_-CT-OS-X  AlkEthOH`

This comes from me -- I added this when trying to create a smirff99Frosst version which would reproduce AlkEthOH energies. However, it looks like I had concluded this did NOT need to be added to smirf99Frosst proper. This seems to have been a mistake, since my notes, pasted above, explicitly say that the C-O generic torsion in smirff99Frosst was not broad enough and did not cover the X-OS-CT-X case. 

So, I think this needs to be re-added AS I did in the AlkEthOH case. 

Let me dump the code I was using to cross-check energies on a github repo so you can re-use it when you start making changes. This is in connection with the SMIRFF/SMIRNOFF paper and probably does not belong in the `smarty` repo.

**Comment by davidlmobley** [comment]
@bannanc - repo shared. Please note this should be valuable for cross-checking against AlkEthOH -- if you make a change you can use the python scripts there to ensure that you are making energies closer to AMBER energetics by making the fix, etc.

**Comment by davidlmobley** [comment]
(It's worth noting that this particular case may actually be an "overly broad" generic which is present in AMBER -- e.g. it may be sort of a "bug" but also avoid the scenario where something gets no torsion at all (in AMBER) or a torsion which is completely generic (in our case). )

**Comment by davidlmobley** [comment]
@bannanc  -- OK, now I'm finally starting to follow your comment about "History of ..." above, which wasn't clear to me at first. You're saying that I broke this by fixing the overly broad generics for CT-CT-OS-CT: 

```
[*:1]-[#6X4:2]-[#8X2H0:3]-[*:4]      1  0.383    0.0  -3.0  parm99 CT-CT-OS-CT
 [*:1]-[#6X4:2]-[#8X2H0:3]-[*:4]      1  0.100  180.0   2.0  parm99 CT-CT-OS-CT
```
> However, I would assume that H-C-O-C and C-C-O-C should not have the same torsion parameters so there was probably an energetic issue before this entry was made too specific

Yes, correct, there was an energetic issue which I "fixed" (potentially incorrectly!) by making these more specific. I did exchange messages with Chris about it but we may have overlooked downstream implications. 

I think you're also right in your later note where you point out that a relevant torsion in one of the intermediate files is this one:
> `[*:1]-[#6X4:2]-[#8X2H0:3]-[*:4]      1  0.383    0.0   3.0  parm99 X_-CT-OS-X  JCC,7,(1986),230`

My take is that this should have been retained and THIS is what's causing the problems, because I saw "CT-CT-OS-CT" expressed with generics and correctly concluded that was incorrect (AND it resulted in energetic problems). Yes, this term happens to be identical to the FIRST term of `CT-CT-OS-CT`, though notice it has only one term (rather than two!) which AMBER denotes by the "3.0" rather than "-3.0" (the minus sign means "there will be another line for this term"). 

You also cite various torsions in parm@frosst but none specifically seem to pertain to this one.

So my tentative conclusion at this point is that we basically need to add the equivalent of this term back in: `[*:1]-[#6X4:2]-[#8X2H0:3]-[*:4]      1  0.383    0.0   3.0  parm99 X_-CT-OS-X  JCC,7,(1986),230`

Does that jive with what you're thinking, @bannanc ?

**Comment by bannanc** [comment]
> So my tentative conclusion at this point is that we basically need to add the equivalent of this term back in: [*:1]-[#6X4:2]-[#8X2H0:3]-[*:4] 1 0.383 0.0 3.0 parm99 X_-CT-OS-X JCC,7,(1986),230

That was also my general conclusion. I will take a look at the repo you shared with me and try adding this back in tomorrow.

**Comment by bannanc** [comment]
@davidlmobley I think I found why you didn't run into this issue while testing.  There are some, possibly important differences between the [smirff99frosst.ffxml file in your testing directory](https://github.com/MobleyLab/AlkEthOH_smirff99Frosst/blob/master/smirff99Frosst.ffxml) and the current version in this repository. 

###### bond "b20" 
in the testing directory: 
`<Bond smirks="[#6X3:1](=[#8X1])-[#8X2:2]" id="b20" k="640.0" length="1.340"/>`
this version: 
`<Bond smirks="[#6X3:1](=[#8X1])-[#8X2H0:2]" id="b20" k="640.0" length="1.340"/>`
This would not have affected test results as there are no carbonyl groups in AlkEthOH 

##### More alcohol torsions in the current version
In the testing directory:
```
<Proper smirks="[*:1]-[#6X4:2]-[#8X2H1:3]-[#1:4]" id="t83" idivf1="1" k1="0.160" periodicity1="3" phase1="0.0" phase2="0.0" k2="0.250" periodicity2="1" idivf2="1"/>
```
In this version, there are two alcohol torsions in this section, the second is more specifically for C-C-O-H:
```
<Proper smirks="[*:1]-[#6X4:2]-[#8X2:3]-[#1:4]" id="t83" idivf1="3" k1="0.500" periodicity1="3" phase1="0.0"/>
    <Proper smirks="[#6X4:1]-[#6X4:2]-[#8X2H1:3]-[#1:4]" id="t84" idivf1="1" k1="0.160" periodicity1="3" phase1="0.0" phase2="0.0" k2="0.250" periodicity2="1" idivf2="1"/>
```

##### Ether torsion is more general
In the test version, the ether torsion discussed above is still more general:
```
<Proper smirks="[*:1]-[#6X4:2]-[#8X2H0:3]-[*:4]" id="t84" idivf1="1" k1="0.383" periodicity1="3" phase1="0.0" phase2="180.0" k2="0.100" periodicity2="2" idivf2="1"/>
```
In this version it is "overly" general
```
<Proper smirks="[#6X4:1]-[#6X4:2]-[#8X2H1:3]-[#1:4]" id="t84" idivf1="1" k1="0.160" periodicity1="3" phase1="0.0" phase2="0.0" k2="0.250" periodicity2="1" idivf2="1"/>
```

**Comment by davidlmobley** [comment]
Hmm, @bannanc -- this is odd. I never manually did anything with that ffxml file, and as far as I know, I only copied the one (generated by script) from in the `smarty` repo both (a) to that directory, and (b) to this repo. So, if there are differences, I would guess they are only version differences. I'll have to dig in to this. Ideally it would be tomorrow, but in all likelihood it wouldn't happen until after CUP.

If you can see the clear path forward though I'm all for you fixing things to the way they "ought" to have been -- ideally we'd end up with something that both (a) reproduces AMBER energies for AlkEthOH (except for the bugs in AMBER!) AND (b) is close to the "condensed" version of parm@Frosst that Christopher intended this to be while also (c) maintaining good coverage.

**Comment by bannanc** [comment]
It seems obvious that we need a more general ether torsion that would type the H-C-O-[not hydrogen] torsions. 
There appear to be two options:

1. from smirffishFrcmod.Torsions.parm99.txt
```
[*:1]-[#6X4:2]-[#8X2H0:3]-[*:4]      1  0.383    0.0   3.0  parm99 X_-CT-OS-X  JCC,7,(1986),230
```
2. from the smirffishFrcmod_AlkEthOH.txt
```
[a,A:1]-[#6X4:2]-[#8X2:3]-[!#1:4]    3  1.150    0.0  -3.0  Frosst X_-CT-OS-X  AlkEthOH
```

I can't find either of these entries in the frcmod.parm_Frosst.

**Comment by bannanc** [comment]
Ok, David explained on slack: 
>`X -CT-OS-X    3    1.15          0.0             3.         JCC,7,(1986),230`
> Note that it uses multiplicity 3; if you take 1.15/3 you get 0.383 which is where we got that entry.

I will include a pull request here and on smarty soon.

**Comment by bannanc** [comment]
This is the ether torsion we fixed in pull request #15
